docs/consume-packages/Overview-and-Workflow.md

```diff
@@ -36,4 +36,54 @@ Occasionally it's necessary to reinstall packages that are already included in a
+
+## What happens when a package is installed?
+
+Simply said, the different NuGet tools typically create a reference to a package in the project file or `packages.config`, then perform a package restore, which effectively installs the package. The exception is `nuget install`, which only expands the package into a `packages` folder and does not modify any other files.
+
+The general process is as follows:
+
+1. (All tools except `nuget.exe`) Record the package identifer and version into the project file or `packages.config`.
+
+2. Acquire the package:
+   - Check if the package (by exact identifer and version number) is already installed in the *global-packages* folder as described on [Managing the global packages and cache folders](managing-the-global-packages-and-cache-folders.md).
+
+   - If the package is not in the *global-packages* folder, attempt to retrieve it from the sources listed listed in the [configuration files](Configuring-NuGet-Behavior.md). For online sources, attempt first to retrieve the package from the cache unless `-NoCache` is specified with `nuget.exe` commands or `--no-cache` is specified with `dotnet restore`. (Visual Studio and `dotnet add package` always use the cache.) If a package is used from the cache, "CACHE" appears in the output. The cache has an expiration time of 30 minutes.
+
+   - If the package is not in the cache, attempt to download it from the sources listed in the configuration. If a package is downloaded, "GET" and "OK" appear in the output.
+
+   - If the package cannot be successfully acquired from any sources, installation fails at this point with an error such as [NU1103](../reference/errors-and-warnings/NU1103.md). Note that errors from `nuget.exe` commands show only the last source checked, but implies that the package wasn't available from any source.
+
+   When acquiring the package, the order of sources in the NuGet configuration may apply:
+
+   - For projects using the PackageReference format, NuGet checks sources local folder and network shares before checking HTTP sources.
+
+   - For projects using the `packages.config` management format, NuGet uses the order of the sources in the configuration. An exception is restore operations, in which case source ordering is ignored and NuGet uses the package from whichever source responds first.
+
+   - In general, the order in which NuGet checks sources isn't particularly meaningful, because any given package with a specific identifier and version number is exactly the same on whatever source it's found.
+
+3. (All tools except `nuget.exe`) Save a copy of the package and other information in the *http-cache* folder as described on [Managing the global packages and cache folders](managing-the-global-packages-and-cache-folders.md).
```

Comment:
this feels like an implementation detail. Why do I, as a package consumer, need to know this?

Also, @nkolev92 - is this true?

> (All tools except `nuget.exe`) Save a copy of the package and other information in the *http-cache* folder